### PR TITLE
Check user agent, and render unsupported browser if it's ie11

### PR DESCRIPTION
**Pull request by mrblippy**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [ ] commit message follows [commit guidelines][]
- [ ] screenshots are included showing significant UI changes

##### Description of change

#1306 

At the moment i am only checking the user agent string for "Trident", to determine if the browser is ie11. The styling has deliberately been kept very simple

ie11 unsupported

Potential issues for discussion:

- Should i also be checking for earlier versions of ie? i personally don't think it's worth it
- Should the error messages be language stringed? 

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/unsupportedbrowser.ftl [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<#include "/com.tle.web.freemarker@/macro/sections/render.ftl"/><#t/>
+<!DOCTYPE html>
+<html>
+<head>
+  <style>
```
Comment (SammyIsConfused) [inline Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/unsupportedbrowser.ftl:5]: Is it just me, or is there some weird formatting happening here? The indenting should be different.

**Review by SammyIsConfused — APPROVED** [review]
Looks good! Just one indentation tweak. 
Also - what happens if you append `?old=true` to the URL? Does it get through then?

**Comment by SammyIsConfused** [comment]
Another note, as discussed in the office; These strings should probably be in the language pack.

File: Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/unsupportedbrowser.ftl [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    html{
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/unsupportedbrowser.ftl:6]:
```suggestion
    html {
```

File: Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/unsupportedbrowser.ftl [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      font-family: Arial, Helvetica, sans-serif;
+      background-color: #FAFAFA;
+      text-align: center;
+      line-height: 1.5em;
+    }
+    #unsupported{
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/unsupportedbrowser.ftl:12]:
```suggestion
    #unsupported {
```

File: Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/unsupportedbrowser.ftl [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      background-color: #FFFFFF;
+      position: absolute;
+      width: 50%;
+      left: 50%;
+      box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2), 0px 1px 1px 0px rgba(0,0,0,0.14), 0px 1px 3px 0px rgba(0,0,0,0.12);
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/unsupportedbrowser.ftl:17]:
```suggestion
      box-shadow: 0px 2px 1px -1px rgba(0, 0, 0, 0.2), 0px 1px 1px 0px rgba(0, 0, 0, 0.14), 0px 1px 3px 0px rgba(0,0,0,0.12);
```

File: Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/unsupportedbrowser.ftl [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      border-radius: 10px;
+      -ms-transform: translate(-50%);
+      margin-top: 24px;
+    }
+  </style>
+</head>
+<div id="unsupported">
+  <h1>Sorry, you are using an unsupported browser.</h1>
+  <p>For the openEQUELLA New UI you must be using a current version of Firefox, Chrome or Microsoft Edge.
+  </p>
```
Comment (ChristianMurphy) [inline Source/Plugins/Core/com.equella.core/resources/view/layouts/outer/unsupportedbrowser.ftl:28]:
```suggestion
  <p>
    For the openEQUELLA New UI you must be using a current version of Firefox, Chrome or Microsoft Edge.
  </p>
```

**Comment by mrblippy** [comment]
> Looks good! Just one indentation tweak.
> Also - what happens if you append `?old=true` to the URL? Does it get through then?

Yep, appending old=true to the url will let them in

**Comment by edalex-ian** [comment]
Just a note based on the formatting comments I'm seeing. Evidentially we don't have formatting for our Freemarker files. So you may want to simply load in IntelliJ and do the <kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>l</kbd>.

**Comment by mrblippy** [comment]
> Just a note based on the formatting comments I'm seeing. Evidentially we don't have formatting for our Freemarker files. So you may want to simply load in IntelliJ and do the CTRL+ALT+l.

Yup, that's the key combo I used to format the ftl file. It's in the next commit

File: Source/Plugins/Core/com.equella.core/scalasrc/com/tle/web/template/RenderNewTemplate.scala [hunk 6]
```diff
@@ -159,6 +160,10 @@ object RenderNewTemplate {
     }
   }
 
+  def isInternetExplorer(request: HttpServletRequest): Boolean = {
+    (Option(request.getHeader("User-Agent")).exists(_.contains("Trident")))
+  }
```
Comment (PenghaiZhang) [inline Source/Plugins/Core/com.equella.core/scalasrc/com/tle/web/template/RenderNewTemplate.scala:165]: This method can be slightly simplified to
```scala
  def isInternetExplorer(request: HttpServletRequest): Boolean = 
    Option(request.getHeader("User-Agent")).exists(_.contains("Trident"))
```

**Review by PenghaiZhang — COMMENTED** [review]
Looks good to me.

File: Source/Plugins/Core/com.equella.core/resources/lang/i18n-resource-centre.properties [hunk 7]
```diff
@@ -4492,3 +4492,5 @@ course.edit.validation.codeinuse=Course with code ''{0}'' already exists.
 impexp.entities=Additional Entities
 api.taxonomy.validation.datasource.unknown=Unknown data source identifier
 api.taxonomy.validation.datasource.notimplemented=Data source identifier not currently implemented
+unsupportedbrowser.heading=Sorry, you are using an unsupported browser.
+unsupportedbrowser.message=For the openEQUELLA New UI you must be using a current version of Firefox, Chrome or Microsoft Edge.
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/resources/lang/i18n-resource-centre.properties:4496]: We've casually discussed the idea o tracking changes to language strings. So, let's start with this PR.

Could you please (as part of this PR) start a new file called `changelog.languagestrings.md` with content like:

```markdown
# Language Strings ChangeLog

This file tracks the changes, additions and deletions to openEQUELLA language strings starting with version 2020.2.
This file can be useful when upgrading, to ensure you make any update to any custom language packs you may have.

## 2020.2

### Modified

- None

### Added

- `unsupportedbrowser.heading`
- `unsupportedbrowser.message`

### Removed

- None
```

**Review by edalex-ian — CHANGES_REQUESTED** [review]
Looks good to me - no issues with what this PR is actually about.

But two things:

1. Could we please target `develop` for this one - it'll make more sense merge wise; and
2. If we can just add the new lang strings changelog that'd be great.

File: Source/Plugins/Core/com.equella.core/resources/lang/i18n-resource-centre.properties (hunk 7 above)
Comment (mrblippy) [reply Source/Plugins/Core/com.equella.core/resources/lang/i18n-resource-centre.properties:4496]: Yep, that's no problem

**Comment by edalex-ian** [comment]
Woah! Hold on a sec, looks like you're request to merge essentially component to develop:

image

**Comment by mrblippy** [comment]
> Woah! Hold on a sec, looks like you're request to merge essentially component to develop:
> 
> image

Yeah, that's bad :/ I will rectify ASAP

**Pull request closed without merging**